File: myec3-socle-webapp-module/myec3-socle-webapp/src/main/resources/org/myec3/socle/webapp/mixins/boundcheckbox.js
/**
* Simple adaptation of
* https://github.com/ioko-tapestry-commons/ioko-tapestry-commons/blob/master/tapestry-commons/tapestry-mixins/src/main/resources/uk/co/ioko/tapestry/mixins/mixins/boundCheckbox.js
*/
var boundCheckBox = Class.create();

boundCheckBox.BoundCheckboxMaster = Class.create({
    initialize: function(clientId) {
            this.childArray = new Array();
            this.element = $(clientId);
            this.element.observe("click", this.onClick.bindAsEventListener(this));
            this.element.__boundcheckbox=this;
    },

    registerChild : function(child) {
    	this.childArray.push(child);
    },

    onClick : function(event) {
        this.childArray.forEach(this.changeCheckBoxes(this.element.checked));
        return false;
    },

    changeCheckBoxes : function(selected) {
        return function(child) {
            child.element.checked = selected;
        }
    }

});

boundCheckBox.BoundCheckboxChild = Class.create({
    initialize: function(clientId, masterId) {
		this.childArray = new Array();
        this.element = $(clientId);
        this.element.observe("click", this.onClick.bindAsEventListener(this));

        this.master = $(masterId);
        this.master.__boundcheckbox.registerChild(this);
    },

    onClick : function(event) {
        if ( this.master.checked && !this.element.checked ) {
        	this.master.checked=false;
        }
        return false;
    }

});

Tapestry.Initializer.boundCheckboxLoad = function(spec) {
	if ( spec.masterId != "" ) {
		new boundCheckBox.BoundCheckboxChild(spec.clientId, spec.masterId);
	} else {
		new boundCheckBox.BoundCheckboxMaster(spec.clientId);
	}
};

File: myec3-socle-webapp-module/myec3-socle-webapp-core/src/main/java/org/myec3/socle/webapp/mixins/BoundCheckBox.java
package org.myec3.socle.webapp.mixins;

import org.apache.tapestry5.ClientElement;
import org.apache.tapestry5.MarkupWriter;
import org.apache.tapestry5.annotations.*;
import org.apache.tapestry5.corelib.components.Checkbox;
import org.apache.tapestry5.json.JSONObject;
import org.apache.tapestry5.services.javascript.JavaScriptSupport;

/**
 * Simple adaptation of
 * https://github.com/ioko-tapestry-commons/ioko-tapestry-commons/blob/master/tapestry-commons/tapestry-mixins/src/main/java/uk/co/ioko/tapestry/mixins/mixins/BoundCheckbox.java
 */
@Import(library = {"boundcheckbox.js"})
public class BoundCheckBox {

    @InjectContainer
    private ClientElement container;

    @Parameter
    private Checkbox master;

    @Environmental
    private JavaScriptSupport renderSupport;

    @AfterRender
    void after(MarkupWriter writer) {
        String masterClientId = master == null ? "" : master.getClientId();
        JSONObject spec = new JSONObject();
        spec.put("clientId", container.getClientId());
        spec.put("masterId", masterClientId);
        renderSupport.addInitializerCall("boundCheckboxLoad", spec);
    }

}
